notebook/notebookapp.py

```diff
@@ -1458,7 +1458,7 @@ def display_url(self):
                 url += '/'
         else:
             if self.ip in ('', '0.0.0.0'):
-                ip = "%s" % socket.gethostname()
+                ip = "%s" % socket.getfqdn()
```

Comment:
I think more needs to happen here.  On my mac, `socket.getfqdn()` returns: `'1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa'` vs. `socket.gethostname()`: `'Kevins-MacBook-Pro.local'` and, while it might be perfectly valid, the former just doesn't make for a good user presentation.  

Also, your title states that you're fixing the value for `custom_display_url`, but this block only occurs when that option has not been configured.

Reply:
I updated the title

Reply:
First, I think a working link is better than a beautiful link not working in other contexts. 

But yes, your display is not user friendly. I don't know what is your local configuration in MAC OS, I am a Linux user, I guess it is the same thing for default configuration?

My vision is that by default, the link should be always valid in local or remote server context, 2 solutions for you:
1. You can configure your `hosts` file to have a better display  for the default value displayed. 
2. Your use case enters in the usage of setting custom_display_url with `Kevins-MacBook-Pro.local` directly.

Reply:
> First, I think a working link is better than a beautiful link not working in other contexts.

I agree. And my Linux machine produces the correct result as well - which also matches the result from `gethostname()`.

I polled a few colleagues and all of their macs return expected values.  I also completely agree with the spirit of this change and had my computer been working like my colleagues, this would have been approved.  

However, while I'm sure I can adjust my configuration to get my scenario working,  the fact that a change is required at all makes me hesitant to accept this PR since I'd be on the hook for answering the onslaught of issues that could potentially arise. Based on the number of google hits I see others encountering, this could be significant - and I suspect most that might have this issue on their computers don't even know it.  (I had never seen an IP like that before testing this out.)

I'm sorry, but I'll need to defer to another maintainer because I'm not interested in signing up for the support burden this could introduce, and the demand for such a change, as best as I can determine, just hasn't been there.